### ISSUE TITLE
`output_changes` incorrectly calculated in the plan file

### ISSUE THREAD

**Issue by marcinwyszynski** [hashicorp/terraform#24628]
### Terraform Version

```
❯ terraform version
Terraform v0.12.24
```

### Terraform Configuration Files
```hcl
provider "random" {}

resource "random_id" "bacon" {
  keepers     = { bacon = "tasty" }
  byte_length = 16
}

output "bacon" { value = random_id.bacon.hex }
```

### Steps to reproduce

After the change has been applied:

```
❯ terraform plan -out repro.plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

random_id.bacon: Refreshing state... [id=eprljZv58YrewNlMCchkQA]

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

❯ terraform plan -out repro.plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

random_id.bacon: Refreshing state... [id=eprljZv58YrewNlMCchkQA]

------------------------------------------------------------------------

No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, no
actions need to be performed.

❯ terraform show -json repro.plan > repro.json
❯ cat repro.json | jq

{
  "format_version": "0.1",
  "terraform_version": "0.12.24",
  "planned_values": {
    "outputs": {
      "bacon": {
        "sensitive": false,
        "value": "7a9ae58d9bf9f18adec0d94c09c86440"
      }
    },
    "root_module": {
      "resources": [
        {
          "address": "random_id.bacon",
          "mode": "managed",
          "type": "random_id",
          "name": "bacon",
          "provider_name": "random",
          "schema_version": 0,
          "values": {
            "b64": "eprljZv58YrewNlMCchkQA",
            "b64_std": "eprljZv58YrewNlMCchkQA==",
            "b64_url": "eprljZv58YrewNlMCchkQA",
            "byte_length": 16,
            "dec": "162970085093216590396273492464954270784",
            "hex": "7a9ae58d9bf9f18adec0d94c09c86440",
            "id": "eprljZv58YrewNlMCchkQA",
            "keepers": { "bacon": "tasty" },
            "prefix": null
          }
        }
      ]
    }
  },
  "resource_changes": [
    {
      "address": "random_id.bacon",
      "mode": "managed",
      "type": "random_id",
      "name": "bacon",
      "provider_name": "random",
      "change": {
        "actions": ["no-op"],
        "before": {
          "b64": "eprljZv58YrewNlMCchkQA",
          "b64_std": "eprljZv58YrewNlMCchkQA==",
          "b64_url": "eprljZv58YrewNlMCchkQA",
          "byte_length": 16,
          "dec": "162970085093216590396273492464954270784",
          "hex": "7a9ae58d9bf9f18adec0d94c09c86440",
          "id": "eprljZv58YrewNlMCchkQA",
          "keepers": { "bacon": "tasty" },
          "prefix": null
        },
        "after": {
          "b64": "eprljZv58YrewNlMCchkQA",
          "b64_std": "eprljZv58YrewNlMCchkQA==",
          "b64_url": "eprljZv58YrewNlMCchkQA",
          "byte_length": 16,
          "dec": "162970085093216590396273492464954270784",
          "hex": "7a9ae58d9bf9f18adec0d94c09c86440",
          "id": "eprljZv58YrewNlMCchkQA",
          "keepers": { "bacon": "tasty" },
          "prefix": null
        },
        "after_unknown": {}
      }
    }
  ],
  "output_changes": {
    "bacon": {
      "actions": ["create"],
      "before": null,
      "after": "7a9ae58d9bf9f18adec0d94c09c86440",
      "after_unknown": false
    }
  },
  "prior_state": {
    "format_version": "0.1",
    "terraform_version": "0.12.24",
    "values": {
      "outputs": {
        "bacon": {
          "sensitive": false,
          "value": "7a9ae58d9bf9f18adec0d94c09c86440"
        }
      },
      "root_module": {
        "resources": [
          {
            "address": "random_id.bacon",
            "mode": "managed",
            "type": "random_id",
            "name": "bacon",
            "provider_name": "random",
            "schema_version": 0,
            "values": {
              "b64": "eprljZv58YrewNlMCchkQA",
              "b64_std": "eprljZv58YrewNlMCchkQA==",
              "b64_url": "eprljZv58YrewNlMCchkQA",
              "byte_length": 16,
              "dec": "162970085093216590396273492464954270784",
              "hex": "7a9ae58d9bf9f18adec0d94c09c86440",
              "id": "eprljZv58YrewNlMCchkQA",
              "keepers": { "bacon": "tasty" },
              "prefix": null
            }
          }
        ]
      }
    }
  },
  "configuration": {
    "provider_config": { "random": { "name": "random" } },
    "root_module": {
      "outputs": {
        "bacon": { "expression": { "references": ["random_id.bacon"] } }
      },
      "resources": [
        {
          "address": "random_id.bacon",
          "mode": "managed",
          "type": "random_id",
          "name": "bacon",
          "provider_config_key": "random",
          "expressions": {
            "byte_length": { "constant_value": 16 },
            "keepers": { "constant_value": { "bacon": "tasty" } }
          },
          "schema_version": 0
        }
      ]
    }
  }
}
```

### Expected Behavior

The resulting JSON should set the value at `$.output_changes.bacon.before` to `"7a9ae58d9bf9f18adec0d94c09c86440"`
The resulting JSON should set the value at `$.output_changes.bacon.actions` to `["no-op"]`

### Actual Behavior

The resulting JSON sets the value at `$.output_changes.bacon.before` to `null`
The resulting JSON incorrectly sets the value at `$.output_changes.bacon.actions` to `["create"]`

**Comment by danieldreier** [comment]
Thanks for reporting this! If I'm reading this right, it looks like you're reporting an issue specific to the `random` provider rather than one with terraform core. I've labeled it, and it will get moved over to that provider's source repository by HashiBot shortly.

Please feel free to re-open this if I'm misunderstanding this and it is in fact a core issue.

**Comment by kmoe** [comment]
I believe this is a known Core issue, not an issue with the `random` provider. For all `output_changes`, no matter the provider, `actions` is always `["create"]`, no matter the value of the output in the state file.

It is [documented](https://www.terraform.io/docs/internals/json-format.html#plan-representation) that "In the Terraform CLI 0.12.0 release, Terraform is not yet fully able to track changes to output values, so the actions indicated may not be fully accurate, but the "after" value will always be correct."

**Comment by skorfmann** [comment]
> I believe this is a known Core issue, not an issue with the `random` provider. For all `output_changes`, no matter the provider, `actions` is always `["create"]`, no matter the value of the output in the state file.

Yes, I just reproduced this in https://github.com/hashicorp/terraform-cdk

**Comment by pselle** [comment]
@skorfmann I made a reproduction of this here, and wasn't able to reproduce on master https://github.com/danieldreier/terraform-issue-reproductions/tree/master/24628 nor on the v0.14 branch. Could you help with a reproduction?

**Comment by kmoe** [comment]
I've tested https://github.com/danieldreier/terraform-issue-reproductions/tree/master/24628 with 0.12.29, 0.13.6, and 0.14.4. The former two versions show:
```
{
  "bacon": {
    "actions": [
      "create"
    ],
    "before": null,
    "after": "3eaf3393733ab75d0a1b567728508d62",
    "after_unknown": false
  }
}
```

and 0.14.4 shows:
```
{
  "bacon": {
    "actions": [
      "no-op"
    ],
    "before": "11f6a28096acfe941a034ed6460b1d02",
    "after": "11f6a28096acfe941a034ed6460b1d02",
    "after_unknown": false
  }
}
```

I therefore believe this bug was fixed somewhere between 0.13.6 and 0.14.4.

I'd still like someone from Core to comment on the [documentation](https://www.terraform.io/docs/internals/json-format.html#plan-representation): "In the Terraform CLI 0.12.0 release, Terraform is not yet fully able to track changes to output values, so the actions indicated may not be fully accurate, but the "after" value will always be correct." Is this still true?
